src/optimum/intel/neural_compressor/config.py

```diff
@@ -0,0 +1,161 @@
+#  Copyright 2021 The HuggingFace Team. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import os
+import requests
+import yaml
+from functools import reduce
+from huggingface_hub import hf_hub_download
+from typing import Any, Optional, Union
+from neural_compressor.conf.config import Quantization_Conf
+
+
+class IncConfig:
+
+    def __init__(
+            self,
+            config_path: str,
+    ):
+        """
+        Args:
+            config_path (:obj:`str`):
+                Path to the YAML configuration file used to control the tuning behavior.
+        Returns:
+            config: neural_compressorConfig object.
+        """
+
+        self.config = Quantization_Conf(config_path)
+
+    def get_config(self, keys: str):
+        return reduce(lambda d, key: d.get(key) if d else None, keys.split("."),
+                      self.config.usr_cfg)
+
+    def set_config(self, keys: str, value: Any):
+        d = self.config.usr_cfg
+        keys = keys.split('.')
+        for key in keys[:-1]:
+            d = d.setdefault(key, {})
+        d[keys[-1]] = value
+
+    @classmethod
+    def from_pretrained(
+            cls,
+            config_name_or_path: Union[str, os.PathLike],
+            config_file_name: str,
+            cache_dir: Optional[Union[str, os.PathLike]] = None,
+            **config_kwargs
+    ):
+        """
+        Instantiate a neural_compressorConfig object from a configuration file which can either be hosted on
```

Comment:
Instantiate an IncConfig object from a configuration file which can either be hosted on huggingface.co or from a local directory path.